python/ray/data/tests/test_parquet.py

```diff
@@ -1176,6 +1176,26 @@ def test_write_num_rows_per_file(tmp_path, ray_start_regular_shared, num_rows_pe
         assert len(table) == num_rows_per_file
 
 
+@pytest.mark.parametrize(
+    "row_data",
+    [
+        [{"a": 1, "b": None}, {"a": 1, "b": 2}],
+        [{"a": None, "b": None}, {"a": 1, "b": 2}],
+        [{"a": 1, "b": 2}, {"a": 1, "b": "hi"}],
```

Comment:
What does the type get promoted to for "b" in this case?

Reply:
Oh - this shouldn't pass actually. it was somehow passing without remove the fusion.